cpp/src/io/orc/stats_enc.cu

```diff
@@ -186,6 +205,15 @@ __device__ inline uint8_t* pb_put_fixed64(uint8_t* p, uint32_t id, void const* r
   return p + 9;
 }
 
+// Splits a nanosecond timestamp into milliseconds and nanoseconds
+__device__ std::pair<int64_t, int32_t> split_nanosecond_timestamp(int64_t nano_count)
```

Comment:
Why are the milliseconds encoded as 64 bit while nanoseconds are 32 bit? Is it because >1e6 ns adds to the ms, whereas the ms can grow unbounded?

Reply:
That's correct. The nanoseconds part is in [0, 999999] range. 
I'm open to suggestions to improve naming/comment here, I also wasn't 100% happy with clarity.